src/hotspot/share/asm/register.hpp

```diff
@@ -245,14 +253,44 @@ inline ReverseRegSetIterator<RegImpl> AbstractRegSet<RegImpl>::rbegin() {
 
 // Debugging support
 
+template<typename R, typename... Rx>
+inline constexpr bool different_registers(AbstractRegSet<R> allocated_regs, R first_register) {
+  return ! allocated_regs.contains(first_register);
+}
+
+template<typename R, typename... Rx>
+inline constexpr bool different_registers(AbstractRegSet<R> allocated_regs, R first_register, Rx... more_registers) {
+  if (allocated_regs.contains(first_register))
+    return false;
+  return different_registers(allocated_regs + first_register, more_registers...);
+}
+
+template<typename R, typename... Rx>
+inline constexpr bool different_registers(R first_register, Rx... more_registers) {
+  return different_registers(AbstractRegSet<R>(first_register), more_registers...);
+}
+
+template<typename R, typename... Rx>
```

Comment:
Rx is unused and not needed.  Similarly for 3-R overload.

Reply:
Isn't it? It seems to me to be used for the next line.

Reply:
I think that comment about unused Rx was on the overload `different_registers(ARS<R>, R)`, which
had an Rx parameter in an old version of this change.  It's okay now.  And there's no longer any 3-R
overload.